docs/migration.rst

```diff
@@ -0,0 +1,249 @@
+    and will be removed in Shapely 2.0. Use the `geoms` property to access the
+    constituent parts of a multi-part geometry.
+    POINT (1 1)
+    POINT (2 2)
+    POINT (3 3)
+
+In Shapely 2.0, all those examples will raise an error.
+
+**How do I update my code?** To access the geometry parts of a multi-part
+geometry, you can use the ``.geoms`` attribute, as the warning indicates.
+
+The examples above can be updated to::
+
+    >>> for part in mp.geoms:
+    ...     print(part)
+    POINT (1 1)
+    POINT (2 2)
+    POINT (3 3)
+    >>> print(mp.geoms[1])
+    POINT (2 2)
+    >>> len(mp.geoms)
+    3
+    >>> list(mp.geoms)
+    [<shapely.geometry.point.Point at 0x7f2e0912bf10>,
+     <shapely.geometry.point.Point at 0x7f2e09fed820>,
+     <shapely.geometry.point.Point at 0x7f2e09fed4c0>]
+
+The single-part geometries (Point, LineString, Polygon) already didn't
+support those features, and for those classes there is no change in behaviour
+for this aspect.
+
+
+The array interface and conversion to NumPy
+===========================================
+
+Shapely provides an array interface to have easy access to the coordinates as,
+for example, NumPy arrays (:ref:`manual section <array-interface>`).
+
+A small example::
+
+    >>> line = LineString([(0, 0), (1, 1), (2, 2)])
+    >>> import numpy as np
+    >>> np.asarray(line)
+    array([[0., 0.],
+           [1., 1.],
+           [2., 2.]])
+
+In addition, there are also the explicit ``array_interface()`` method and
+``ctypes`` attribute to get access to the coordinates as array data:
+
+    >>> line.ctypes
+    <shapely.geometry.linestring.c_double_Array_6 at 0x7f75261eb740>
+    >>> line.array_interface()
+    {'version': 3,
+     'typestr': '<f8',
+     'data': <shapely.geometry.linestring.c_double_Array_6 at 0x7f752664ae40>,
+     'shape': (3, 2)}
+
+This functionality is available for Point, LineString, LinearRing and MultiPoint.
+
+For more robust interoperability with NumPy, this array interface will be removed
+from those geometry classes, and limited to the ``coords``. 
+
+Starting with Shapely 1.8, converting a geometry object to a NumPy array
+directly will start raising a warning::
+
+    >>> np.asarray(line)
+    ShapelyDeprecationWarning: The array interface is deprecated and will no longer
+    work in Shapely 2.0. Convert the '.coords' to a NumPy array instead.
+    array([[0., 0.],
+           [1., 1.],
+           [2., 2.]])
+
+**How do I update my code?** To convert a geometry to a NumPy array, you can
+convert the ``.coords`` attribute instead::
+
+    >>> line.coords
+    <shapely.coords.CoordinateSequence at 0x7f2e09e88d60>
+    >>> np.array(line.coords)
+    array([[0., 0.],
+           [1., 1.],
+           [2., 2.]])
+
+The ``array_interface()`` method and ``ctypes`` attribute will be removed in
+Shapely 2.0, but since Shapely will start requiring NumPy as a dependency,
+you can use NumPy or its array interface directly.
+
+
```

Comment:
Another related issue is creating an object array of shapely geometries, described in #1096. Is it worth discussing? It only impacts version 1.8.

Reply:
Yes, I certainly think that's worth describing (see also my comment https://github.com/Toblerity/Shapely/pull/1180#issuecomment-903147126 above). Will add a section about that.